File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> hm;

        for (int i = 0; i < n; i++){
            int diff = target - nums[i];
            if (hm.find(diff) != hm.end()){
                return {hm[diff], i};
            }
            hm[nums[i]] = i;
        }
        return {};
    }
};

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l = 0
        curr = 0
        ans = float('inf')
        for r in range(len(nums)):
            curr += nums[r]
            while curr >= target:
                ans = min(ans, r - l + 1)
                curr -= nums[l]
                l += 1

        return 0 if ans == float('inf') else ans
